flytekit/bin/entrypoint.py

```diff
@@ -584,6 +623,8 @@ def handle_sigterm(signum, frame):
 @click.option("--resolver", required=True)
 @click.option("--checkpoint-path", required=False)
 @click.option("--prev-checkpoint", required=False)
+@click.option("--pickled", is_flag=True, default=False, help="Use this to mark if the distribution is pickled.")
```

Comment:
I think it's better to add a pickle resolver and add it to the compilation context [here](https://github.com/flyteorg/flytekit/blob/dfe93eb55fadf26190e898a25936dad0d5deaf39/flytekit/core/python_auto_container.py#L119), so that we don't need to change the entrypoint.

Reply:
I modified the behavior of the default resolver so that it could support the pickled task.